### PR TITLE
Following feature

### PR DESCRIPTION
**Pull request by eduardopoleo**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -18,6 +18,10 @@ def require_user
     end
   end
 
+  def number_of_followers(user)
```
Comment (brandoncc) [inline app/controllers/application_controller.rb]: This seems like it should be a User instance method.

File: spec/controllers/followings_controller_spec.rb [hunk 2]
```diff
@@ -24,6 +24,118 @@
...
     end
 
-    it 'redirects to the singin path if the user is not authenticated'
+    it 'orders the followings such that last followings shows first' do
+      jose = Fabricate(:user)
+      carlos = Fabricate(:user)
+      following1 = Fabricate(:following, user_id: alice.id, subject_id: jose.id)
+      following2 = Fabricate(:following, user_id: alice.id, subject_id: carlos.id)
+      get :index, user_id: alice.id
+      expect(assigns(:followings)).to match_array([following2, following1])
```
Comment (brandoncc) [inline spec/controllers/followings_controller_spec.rb]: `match_array` does not check order, only that the elements exist. The description of this spec does not match this expectation.

File: spec/controllers/followings_controller_spec.rb [hunk 3]
```diff
@@ -24,6 +24,118 @@
...
+      let(:user) {Fabricate(:user)}
+      let(:action) {post :create, user_id: alice.id, subject_id: user.id} 
+    end
+  end
+
+  describe 'DELETE destroy' do
+    let(:alice) {Fabricate(:user)}
+    before{session[:user_id] = alice.id}
+
+    it 'sets the following to be destroyed' do
```
Comment (brandoncc) [inline spec/controllers/followings_controller_spec.rb]: The next spec technically can't pass if this one fails....this one is a bit of overkill in my opinion. You should only test for instance variables when they will be used in a view.

File: spec/features/user_follows_another_user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    unfollow 
+    expect(alice.subjects.size).to eq(0)
+  end
+
+  def follow(user)
+    find("a[href='/followings?subject_id=#{user.id}']").click
+  end
+
+  def unfollow 
+    find("a[href='/followings/#{Following.first.id}']").click
```
Comment (brandoncc) [inline spec/features/user_follows_another_user_spec.rb:28]: This method is pretty tightly coupled to the spec it is used in. That limits the usefulness of the method.

Comment (eduardopoleo) [inline spec/features/user_follows_another_user_spec.rb:28]: Hey Brandon, I am not really sure what to do here. Should I just include the 
find("a[href='/followings/#{Following.first.id}']").click
into the spec without extracting it away in a method.
I was just trying to keep the language used in the spec as human readable as possible, as suggested in the lectures.

Comment (brandoncc) [inline spec/features/user_follows_another_user_spec.rb:28]: I would add a parameter so you can pass in the following or id (whichever you prefer), and pass in `Following.first`. That will allow you to use the method but will make the method more versatile.

**Pull request closed without merging**